modules/programs/afew.nix

```diff
@@ -6,21 +6,134 @@ let
 
   cfg = config.programs.afew;
 
+  dag = config.lib.dag;
+
+  mkIniKeyValue = key: value:
+    let
+      tweakVal = v:
+        if isString v then v
+        else if isList v then concatMapStringsSep ";" tweakVal v
+        else if isBool v then (if v then "true" else "false")
+        else toString v;
+    in
+      "${key}=${tweakVal value}";
+
+  mkDag = filters:
+    let
+      mkNode = nodeName: filter:
+        let
+          before = (filter.before or cfg.defaultBefore);
+          after = (filter.after or cfg.defaultAfter);
+          filterName = (filter.class or "Filter");
+          filterKV = (removeAttrs filter ["after" "before" "class"]) // {
+            "#nodeName" = nodeName;
+            "#before" = before;
+            "#after" = after;
+          };
+        in
+          dag.entryBetween before after (nameValuePair filterName filterKV);
+    in
+      mapAttrs mkNode filters;
+
+  /* Return a topologically sorted list of DAG entries encapsulating
+     name value pairs.
+  */
+  reorderFilters = filters:
+    (dag.topoSort (mkDag filters)).result;
+
+  /* Convert a topologically sorted list of DAG nodes to a list of INI
+     sections.  The section name is appended with an index number to
+     prevent clashes, as is required by afew.
```

Comment:
Is the index number still necessary? Seems to me that the attribute name is used now and that should by definition be unique.

Reply:
The index number is used to prevent clashes. See [here](https://afew.readthedocs.io/en/latest/filters.html#headermatchingfilter) for an example in afew documentation. Unfortunately I don't think we can get rid of it as long as a few requires it.